datalad/distribution/update.py

```diff
@@ -226,6 +233,13 @@ def __call__(
             reobtain_data=False):
         if fetch_all is not None:
             lgr.warning('update(fetch_all=...) called. Option has no effect, and will be removed')
+        if sibling is not None:
+            import warnings
+            warnings.warn("The --sibling parameter has been renamed to -s/"
+                          "--sibling-name.",
```

Comment:
I believe similar question came, and the context is related to #6213 -- `--sibling` is how we describe it in CLI, whenever in Python API would be `sibling`.

<details>
<summary>we don't have yet consistency (a sample grep showing another effect of absent centralized helper - duplication)</summary> 

```shell
datalad/core/local/status.py-            warnings.warn(
datalad/core/local/status.py-                "status(repor_filetype=) no longer supported, and will be removed "
datalad/core/local/status.py-                "in a future release",
datalad/core/local/status.py:                DeprecationWarning)
--
datalad/distributed/create_sibling_github.py-                "datalad-create-sibling-github's `dryrun` option is "
datalad/distributed/create_sibling_github.py-                "deprecated and will be removed in a future release, "
datalad/distributed/create_sibling_github.py-                "use the renamed `dry_run/--dry-run` option instead.",
datalad/distributed/create_sibling_github.py:                DeprecationWarning)
--
datalad/distributed/create_sibling_github.py-                "datalad-create-sibling-github's `github_login` option is "
datalad/distributed/create_sibling_github.py-                "deprecated and will be removed in a future release, "
datalad/distributed/create_sibling_github.py-                "use the `credential` option instead.",
datalad/distributed/create_sibling_github.py:                DeprecationWarning)
--
datalad/distributed/create_sibling_github.py-                "datalad-create-sibling-github's `github_organization` "
datalad/distributed/create_sibling_github.py-                "option is deprecated and will be removed in a future "
datalad/distributed/create_sibling_github.py-                "release, prefix the reposity name with `<org>/` instead.",
datalad/distributed/create_sibling_github.py:                DeprecationWarning)
--
datalad/distributed/create_sibling_gitlab.py-                "datalad-create-sibling-github's `dryrun` option is "
datalad/distributed/create_sibling_gitlab.py-                "deprecated and will be removed in a future release, "
datalad/distributed/create_sibling_gitlab.py-                "use the renamed `dry_run/--dry-run` option instead.",
datalad/distributed/create_sibling_gitlab.py:                DeprecationWarning)
--
datalad/distributed/create_sibling_ria.py-            import warnings
datalad/distributed/create_sibling_ria.py-            warnings.warn("datalad-create-sibling-ria --no-storage-sibling "
datalad/distributed/create_sibling_ria.py-                          "is deprecated, use --storage-sibling off instead.",
datalad/distributed/create_sibling_ria.py:                          DeprecationWarning)
--
datalad/distributed/drop.py-            warnings.warn(
datalad/distributed/drop.py-                "The `check` argument of `datalad drop` is deprecated, "
datalad/distributed/drop.py-                "use the `reckless` argument instead.",
datalad/distributed/drop.py:                DeprecationWarning)
--
datalad/distributed/drop.py-                "The `if_dirty` argument of `datalad drop` is ignored, "
datalad/distributed/drop.py-                "it can be removed for a safe-by-default behavior. For "
datalad/distributed/drop.py-                "other cases consider the `reckless` argument.",
datalad/distributed/drop.py:                DeprecationWarning)

```
</details>

so may be following the `create_sibling*` trend, make it `sibling/--sibling` and `sibling_name/--sibling-name` for now?

Reply:
I made a concrete proposal for a uniform specification in https://github.com/datalad/datalad/pull/6249

It also implements support for API-specific doc formatting for it. I went for `||` instead of `/`, because the latter is sometimes used for alternatives within an API (e.g. long v short), and I went for `,` to separate long v short, because that is what the manpages do.